src/Swift.Runtime/src/Swift/SwiftString.cs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Swift;
+using System.Text;
+using Swift.Runtime;
+using Swift.Runtime.InteropServices;
+
+namespace Swift;
+
+/// <summary>
+/// Represents Foundation.Data type.
+/// </summary>
+public struct Data
+{
+    private long _flags;
+    private IntPtr _object;
+}
+
+/// <summary>
+/// Represents a Swift string with Foundation.Data payload.
+/// </summary>
+public struct SwiftString : ISwiftObject
+{
+    static nuint _payloadSize = SwiftObjectHelper<SwiftString>.GetTypeMetadata().Size;
+
+    private Data _payload;
+
+    private static Dictionary<Type, string> _protocolConformanceSymbols;
+
+    static SwiftString()
+    {
+        _protocolConformanceSymbols = new Dictionary<Type, string> { };
+    }
+
+    public static nuint PayloadSize => _payloadSize;
```

Comment:
Just curious - why are we exposing these as public? I would expect the projection to "hide" the specifics of the payload and calling conv for me. And unless we have a specific need, we should hide it.

Reply:
Those are public now because of the design of the `MarshalToSwift` API. We need to know how big something is to be able to allocate enough space before calling `MarshalToSwift`. Once we redesign this API this can become private.